### PR TITLE
Elliot/circles 348 - planned/unplanned planner migration to backend

### PR DESCRIPTION
**Pull request by sixohtoo**

### CODE REVIEW

**Comment by Peedee2002** [comment]
merge conflicts go brr

**Comment by Peedee2002** [comment]
Mypy failed, you should install a mypy plugin if you use vscode or steal @imagine-hussain's config if youre using vim

**Comment by imagine-hussain** [comment]
> Mypy failed, you should install a mypy plugin if you use vscode or steal @imagine-hussain's config if youre using vim

also, u can just run `mypy ./backend` or `cd backend; mypy .` from the terminal to get the output too

File: backend/server/routers/planner.py [hunk 1]
```diff
@@ -57,7 +67,217 @@ def validate_term_planner(plannerData: PlannerData):
...
+        planner['years'][data.destRow][data.destTerm].insert(data.destIndex, data.courseCode)
+    set_user(token, user, True)
+
+@router.post("/plannedToTerm")
+def set_planned_course_to_term(data: PlannedToTerm, token: str = DUMMY_TOKEN):
+    course = get_course(data.courseCode)
+    user = get_user(token)
+    planner = user['planner']
+
+    uoc, terms_offered, is_multiterm = itemgetter('UOC', 'terms', 'is_multiterm')(course)
```
Comment (Peedee2002) [inline backend/server/routers/planner.py:153]: itemgetter is a crazy function lol

Comment (Peedee2002) [reply backend/server/routers/planner.py:153]: I do like it tho

File: backend/server/routers/planner.py [hunk 2]
```diff
@@ -57,7 +67,217 @@ def validate_term_planner(plannerData: PlannerData):
...
+        for term_row in terms_list:
+            term, row_offset = itemgetter('term', 'row_offset')(term_row)
+            index = len(planner['years'][data.destRow + row_offset][term])
+            planner['years'][data.destRow + row_offset][term].insert(index, data.courseCode)
+    else:
+        planner['years'][data.destRow][data.destTerm].insert(data.destIndex, data.courseCode)
+    set_user(token, user, True)
+
+@router.post("/plannedToTerm")
+def set_planned_course_to_term(data: PlannedToTerm, token: str = DUMMY_TOKEN):
```
Comment (Peedee2002) [inline backend/server/routers/planner.py:130]: this function gives me a headache, but idk how to improve it

**Review by Peedee2002 — APPROVED** [review]
I wanted someone else to review but it's ok

**Pull request merged**